tests/src/Functional/ContentEventRenderTest.php

```diff
@@ -210,6 +213,91 @@ public function testEventRenderingTeaser(): void {
     ];
     $card_assert->assertVariant('search', $html);
     $card_assert->assertPattern($expected, $html);
+
+    // Test timezone on event teaser.
+    // Set an explicit site timezone: UTC.
+    $this->config('system.date')->set('timezone.default', 'UTC')->save();
+
+    // Values stored are UTC.
+    $node->set('oe_sc_event_dates', [
+      'value' => '2024-06-07T00:00:00',
+      'end_value' => '2024-06-07T23:59:59',
+    ]);
+    $node->save();
+
+    // The site timezone is UTC so the event time is the same day.
+    $expected['content'] = [
+      '7 Jun 2024',
+      'Brussel, Belgium',
+    ];
+    $expected['date'] = [
+      'year' => '2024',
+      'month' => 'Jun',
+      'day' => '07',
+      'end_year' => '2024',
+      'date_time' => '2024-06-07',
+    ];
+
+    $build = $builder->view($node, 'teaser');
+    $html = (string) $this->container->get('renderer')->renderRoot($build);
+
+    $card_assert->assertVariant('search', $html);
+    $card_assert->assertPattern($expected, $html);
+
+    // Check date rendering based on user timezone.
+    // Set an explicit user timezone: UTC -5.
+    $test_user = $this->createUser([], NULL, FALSE, ['timezone' => 'America/New_York']);
+    $this->drupalLogin($test_user);
+
+    // The user timezone is -5 so the event time be five hours earlier and will
+    // start the day before.
+    $expected['content'] = [
+      '6 Jun 2024 - 7 Jun 2024',
+      'Brussel, Belgium',
+    ];
+    $expected['date'] = [
+      'year' => '2024',
+      'month' => 'Jun',
+      'day' => '06',
+      'end_day' => '07',
+      'end_month' => 'Jun',
+      'end_year' => '2024',
+      'date_time' => '2024-06-06',
+    ];
+
+    $build = $builder->view($node, 'teaser');
+    $html = (string) $this->container->get('renderer')->renderRoot($build);
+
+    $card_assert->assertVariant('search', $html);
+    $card_assert->assertPattern($expected, $html);
+    $this->drupalLogout();
+
+    // Check date rendering based on site timezone.
+    // Set an explicit site timezone: UTC +2.
+    $this->config('system.date')->set('timezone.default', 'Europe/Madrid')->save();
+
+    // The site timezone is +2 so the event time be two hours later and will
+    // last until the next day.
+    $expected['content'] = [
+      '7 Jun 2024 - 8 Jun 2024',
+      'Brussel, Belgium',
+    ];
+    $expected['date'] = [
+      'year' => '2024',
+      'month' => 'Jun',
+      'day' => '07',
+      'end_day' => '08',
+      'end_month' => 'Jun',
+      'end_year' => '2024',
+      'date_time' => '2024-06-07',
+    ];
+
+    $build = $builder->view($node, 'teaser');
+
+    $html = (string) $this->container->get('renderer')->renderRoot($build);
+
+    $card_assert->assertVariant('search', $html);
+    $card_assert->assertPattern($expected, $html);
```

Comment:
Lets do 1 more thing, disable the user configurabele timezone, check that even though the user has a different time zone, the user still sees the sites timezone